### PR TITLE
QuickEditor: New OpenAPI spec

### LINKED ISSUE #290: QuickEditor: Import the new OpenAPI spec

### PR DESCRIPTION
**Pull request by AdamGrzybkowski**
Closes #290

### Description

This PR:
1. Updates the OpenAPI spec
2. Regenerates the code
3. Updates the QE code to use new/updated endpoints. This includes:
- To fetch avatars we no longer need the `/identities` endpoint. Simplified to one only.
- When uploading the image, we do get the `Avatar` object in the response, meaning we can skip the extra request to refresh the avatar list. 

### Testing Steps

Smoke test the QE, especially avatar upload.

### CODE REVIEW

File: gravatar/src/main/java/com/gravatar/restapi/models/Error.kt [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ *
+ */
+package com.gravatar.restapi.models
+
+import com.google.gson.annotations.SerializedName
+import java.util.Objects
+
+/**
+ *
+ *
```
Comment (hamorillo) [inline gravatar/src/main/java/com/gravatar/restapi/models/Error.kt:15]: ⛏ Not a direct issue of the PR, but we can add the class description in the specification file.

File: gravatar-quickeditor/src/main/java/com/gravatar/quickeditor/ui/avatarpicker/AvatarPickerUiState.kt [hunk 2]
```diff
@@ -12,7 +12,7 @@ internal data class AvatarPickerUiState(
     val isLoading: Boolean = false,
     val error: SectionError? = null,
     val profile: ComponentState<Profile>? = null,
-    val identityAvatars: IdentityAvatars? = null,
+    val identityAvatars: EmailAvatars? = null,
```
Comment (hamorillo) [inline gravatar-quickeditor/src/main/java/com/gravatar/quickeditor/ui/avatarpicker/AvatarPickerUiState.kt]: ⛏ It's an internal class, but I think it's better to eliminate all `identity` naming. There are more `identity`s in other classes.

File: gravatar-quickeditor/src/test/java/com/gravatar/quickeditor/data/repository/AvatarRepositoryTest.kt [hunk 3]
```diff
@@ -54,51 +50,34 @@ class AvatarRepositoryTest {
...
-
-        assertEquals(
-            Result.Failure<IdentityAvatars, QuickEditorError>(QuickEditorError.Request(ErrorType.SERVER)),
-            result,
-        )
-    }
-
-    @Test
-    fun `given token stored when avatar service and identity service succeed then Success result`() = runTest {
+    fun `given token stored when avatar service succeed then Success result`() = runTest {
```
Comment (hamorillo) [inline gravatar-quickeditor/src/test/java/com/gravatar/quickeditor/data/repository/AvatarRepositoryTest.kt]: ⛏ We can specify the action in the method declaration. Maybe, something like:

```suggestion
    fun `given token stored when fetch avatars succeed then Success result`() = runTest {
```

**Review by hamorillo — COMMENTED** [review]
🛑 We've removed the UCrop activity declaration [here](https://github.com/Automattic/Gravatar-SDK-Android/pull/315/commits/eef2e1aab426549d0dad55ff54cb8d88642a33d0) and we need to add it in the quickeditor module manifest. 

```
android.content.ActivityNotFoundException: Unable to find explicit activity class {com.gravatar.demoapp/com.yalantis.ucrop.UCropActivity}; have you declared this activity in your AndroidManifest.xml?
```

**Edited:** This is not a concrete issue of this PR.

⚠️ I saw [this comment](https://github.com/Automattic/Gravatar-SDK-Android/pull/265#issuecomment-2298850357) where we said the scroll should move to the first avatar when a new one is uploaded. However it's not working for me. If I'm at the end of the horizontal scroll I can't see the new element.

**Comment by AdamGrzybkowski** [comment]
>We've removed the UCrop activity declaration [here](https://github.com/Automattic/Gravatar-SDK-Android/pull/315/commits/eef2e1aab426549d0dad55ff54cb8d88642a33d0) and we need to add it in the quickeditor module manifest.


Let's fix this in a separate PR to not block it by this big one.

EDIT: [PR ](https://github.com/Automattic/Gravatar-SDK-Android/pull/322)

**Comment by AdamGrzybkowski** [comment]
>⚠️ I saw https://github.com/Automattic/Gravatar-SDK-Android/pull/265#issuecomment-2298850357 where we said the scroll should move to the first avatar when a new one is uploaded. However it's not working for me. If I'm at the end of the horizontal scroll I can't see the new element.


Does it work sometimes or never? Have you checked `trunk`?

File: gravatar-quickeditor/src/main/java/com/gravatar/quickeditor/ui/avatarpicker/AvatarPickerUiState.kt (hunk 2 above)
Comment (AdamGrzybkowski) [reply gravatar-quickeditor/src/main/java/com/gravatar/quickeditor/ui/avatarpicker/AvatarPickerUiState.kt]: Yes, good catch.

File: gravatar/src/main/java/com/gravatar/restapi/models/Error.kt (hunk 1 above)
Comment (AdamGrzybkowski) [reply gravatar/src/main/java/com/gravatar/restapi/models/Error.kt:15]: I've asked Diego about this.

**Comment by hamorillo** [comment]
> Does it work sometimes or never? Have you checked trunk?

It works in the first upload but not in the following ones:

https://github.com/user-attachments/assets/04ab0863-fb41-4f9d-bda3-937862c602f3

On `trunk`, it always scrolls to the first element.

https://github.com/user-attachments/assets/209d451d-d39e-4280-9311-12f79de59666

**Comment by AdamGrzybkowski** [comment]
Thanks @hamorillo! The issue should be fixed now.

File: gravatar-quickeditor/src/main/java/com/gravatar/quickeditor/ui/avatarpicker/AvatarPickerViewModel.kt [hunk 4]
```diff
@@ -116,6 +116,7 @@ internal class AvatarPickerViewModel(
                                     addAll(currentState.emailAvatars.avatars.filter { it.imageId != avatar.imageId })
                                 },
                             ),
+                            scrollToIndex = null,
```
Comment (hamorillo) [inline gravatar-quickeditor/src/main/java/com/gravatar/quickeditor/ui/avatarpicker/AvatarPickerViewModel.kt:119]: ⚠️ We need to do the same in the Failure case. 

https://github.com/user-attachments/assets/22badcc1-61e0-4f8e-adad-bae7aadbd5bc

Comment (AdamGrzybkowski) [reply gravatar-quickeditor/src/main/java/com/gravatar/quickeditor/ui/avatarpicker/AvatarPickerViewModel.kt:119]: right... fixed!

**Review by hamorillo — APPROVED** [review]
Works great! Thank for the changes! 🚀

**Pull request merged**